### PR TITLE
Add kraken ohlcv unit tests

### PR DESCRIPTION
**Pull request by kobejean**

### CODE REVIEW

File: crypto_analytics/collection/data_source/kraken.py [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
 from crypto_analytics.collection.data_source import OHLCVDataSource
 from crypto_analytics.types import Interval
 
 class KrakenOHLCV(OHLCVDataSource):
     columns = ['time', 'open', 'high', 'low', 'close', 'vwap', 'volume', 'count']
+    # TODO: define appropriate dtypes
+    dtypes = {'time': np.int64, 'open': object, 'high': object, 'low': object, 'close': object, 'vwap': object, 'volume': object, 'count': np.int64 }
 
-    def __init__(self, interval: Interval, pair: str, since: int = None):
+    def __init__(self, interval: Interval, pair: str, rows: int, last_time: int = None):
```
Comment (kobejean) [inline crypto_analytics/collection/data_source/kraken.py:15]: I'm trying to move towards unifying common data source params. Eventually we should have some sort of `SymbolPair` type but I think `rows` and `last_time` might be common data source parameters that'll help me with writing a collection script.

File: crypto_analytics/collection/data_source/kraken.py [hunk 2]
```diff
@@ -25,39 +29,57 @@ def fetch(self) -> pd.DataFrame:
...
+
+        self.__validate_data(data, response)
+
+        self.data = data
         return self.data
 
 
     def write(self, filepath: str):
-        self.data.to_csv(filepath)
+        self.data.to_csv(filepath, index=False)
```
Comment (kobejean) [inline crypto_analytics/collection/data_source/kraken.py:58]: I realized that we should be using `index=False` for our write methods. I'll add them to our other classes later.

File: tests/unit_tests/collection/test_k_ohclv.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+from crypto_analytics.types import Interval
+
+# mock data
+
+k_ohclv_dtypes = {'time': np.int64, 'open': object, 'high': object, 'low': object, 'close': object, 'vwap': object, 'volume': object, 'count': np.int64 }
+k_ohclv_columns = ['time', 'open', 'high', 'low', 'close', 'vwap', 'volume', 'count']
+k_ohclv_success = {'error': [], 'result': {'XXBTZUSD': [[1560123060, '7633.2', '7636.2', '7633.2', '7635.6', '7635.7', '2.23099305', 6]], 'last': 1560123060}}
+k_ohclv_success_df = pd.DataFrame(k_ohclv_success['result']['XXBTZUSD'], columns=k_ohclv_columns).astype(k_ohclv_dtypes)
+k_ohclv_incomplete_candle = {'error': [], 'result': {'XXBTZUSD': [[1560123060, '7633.2', '7636.2', '7633.2', '7635.6', '7635.7', '2.23099305', 6], [1560123120, '7635.6', '7635.6', '7630.5', '7633.1', '7630.6', '0.58258092', 2]], 'last': 1560123060}}
+k_ohclv_incomplete_candle_df = pd.DataFrame(k_ohclv_incomplete_candle['result']['XXBTZUSD'], columns=k_ohclv_columns).astype(k_ohclv_dtypes)
```
Comment (kobejean) [inline tests/unit_tests/collection/test_k_ohclv.py:16]: Importing mock data from `mock_data.py` was getting ridiculous. I think it's better to keep all our test case info on the same file anyway.

File: tests/unit_tests/collection/test_k_ohclv.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    candles.data = k_ohclv_success_df
+    # when
+    data = candles.get_volume()
+    # then
+    expected_data = k_ohclv_success_df['volume']
+    assert_series_equal(data, expected_data)
+
+# write method tests
+
+def test_k_ohlcv_write(requests_mock, tmp_dir):
```
Comment (kobejean) [inline tests/unit_tests/collection/test_k_ohclv.py:157]: If you're writing unit tests for there `write()` methods you can use this `tmp_dir` fixture to write to a temporary directory.

Comment (kobejean) [reply tests/unit_tests/collection/test_k_ohclv.py:157]: They should work anywhere in our `tests` directory

File: tests/unit_tests/collection/test_k_ohclv.py (hunk 3 above)
Comment (kobejean) [reply tests/unit_tests/collection/test_k_ohclv.py:16]: I'm eventually going to remove the `mock_data.py` file but I think that's out of the scope of this story.

File: crypto_analytics/collection/data_source/kraken.py (hunk 2 above)
Comment (Jacoab) [reply crypto_analytics/collection/data_source/kraken.py:58]: I can take care of that today.

**Review by Jacoab — APPROVED** [review]

**Pull request merged**